packages/circuit-ui/components/Popover/Popover.tsx

```diff
@@ -0,0 +1,267 @@
+/**
+ * Copyright 2021, SumUp Ltd.
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/** @jsx jsx */
+import { css, jsx } from '@emotion/core';
+import { Theme } from '@sumup/design-tokens';
+import React, {
+  useState,
+  FC,
+  HTMLProps,
+  ReactNode,
+  SVGProps,
+  MouseEvent,
+  Fragment,
+  useMemo,
+  RefObject,
+} from 'react';
+import { useClickAway, useLatest } from 'react-use';
+import { Dispatch as TrackingProps } from '@sumup/collector';
+import { usePopper } from 'react-popper';
+import { Placement, State, Modifier } from '@popperjs/core';
+import { useTheme } from 'emotion-theming';
+
+import styled, { StyleProps } from '../../styles/styled';
+import { listItem, textMega } from '../../styles/style-mixins';
+import { useComponents } from '../ComponentsContext';
+import useClickHandler from '../../hooks/use-click-handler';
+import Hr from '../Hr';
+
+export interface BaseProps {
+  children: ReactNode;
+  /**
+   * Display an icon in addition to the label.
+   */
+  icon?: FC<SVGProps<SVGSVGElement>>;
+
+  destructive?: boolean;
+  /**
+   * Additional data that is dispatched with the tracking event.
+   */
+  tracking?: TrackingProps;
+  /**
+   * The ref to the html dom element, it can be a button an anchor or a span, typed as any for now because of complex js manipulation with styled components
+   */
+  ref?: React.Ref<HTMLButtonElement & HTMLAnchorElement>;
+}
+
+type LinkElProps = Omit<HTMLProps<HTMLAnchorElement>, 'size' | 'type'>;
+type ButtonElProps = Omit<HTMLProps<HTMLButtonElement>, 'size' | 'type'>;
+
+export type PopoverItemProps = BaseProps & LinkElProps & ButtonElProps;
+type PopoverItemWrapperProps = LinkElProps & ButtonElProps;
+
+const itemWrapperStyles = () => css`
+  label: popover-item;
+  display: flex;
+  justify-content: center;
+  align-items: flex-start;
+`;
+
+const PopoverItemWrapper = styled('button')<PopoverItemWrapperProps>(
+  listItem,
+  itemWrapperStyles,
+  textMega,
+);
+
+const iconStyles = (theme: Theme) => css`
+  label: popover__icon;
+  margin-right: ${theme.spacings.byte};
+`;
+
+export const PopoverItem = ({
```

Comment:
I would extract this component (and the related specs, types, styles...) into a separate component directory

Reply:
I felt that a separate file wasn't necessary. PopoverItem an integral part of the component, it's small, and keeping the code in one file makes it easier to read and reduces boilerplate.

(It's a question of conventions vs convenience. I've been leaning more towards convenience lately)

Reply:
For me it was about readability, I got quite confused with the Popover, PopoverItem, PopoverWrapper, PopoverItemWrapper, all of their styles and their types in the same file 😛

Reply:
I don't feel strongly about it. @amelako, what do you think?

Reply:
I would rather leave it as well within the Popover component. Maybe we could think only about renaming it, so it is not confusing? Wdyt about that?